uptane-standard.md

```diff
@@ -98,83 +98,128 @@ TODO
 
-### Delegated Targets Roles
+The Image repository SHALL implement storage which permits authorized users to write an image file using a unique filename, and later read the same file using the same name. It MAY use any filesystem, key-value store, or database that fulfills this requirement.
 
-#### The Supplier Roles
+The Image repository MAY require authentication for read access.
 
-### The Snapshot Role
+### Director Repository
+
+The Director repository instructs ECUs as to which images should be installed by producing signed metadata on demand. Unlike the Image repository, it is mostly controlled by automated, online processes. It also consults a private inventory database containing information on vehicles, ECUs, and software revisions.
+
+The Directory repository SHALL expose an interface for primaries to upload vehicle version manifests and download metadata. This interface SHOULD be public.
+The Director MAY encrypt images for ECUs that require it, either by encrypting on-the-fly or by storing encrypted images in the repository.
 
-### The Timestamp Role
+The Director repository SHALL implement storage which permits an automated service to write generated metadata files. It MAY use any filesystem, key-value store, or database that fulfills this requirement.
 
-### The Map File
+#### Directing installation of images on vehicles
 
-# Metadata
+A Director repository MUST conform to the following six-step process for directing the installation of software images on a vehicle.
 
-## Common Metadata Structures and Formats
+1. When the Director receives a vehicle version manifest sent by a primary (as described in {{construct_manifest}}), it decodes the manifest, and determines the unique vehicle identifier.
+1. Using the vehicle identifier, the Director queries its inventory database (as described in {{inventory_db}}) for relevant information about each ECU in the vehicle.
+1. The Director checks the manifest for accuracy compared to the information in the inventory database. If any of the required checks fail, the Director drops the request. An implementor MAY make whatever additional checks they wish. At a minimum, the following checks are required:
+    * Each ECU recorded in the inventory database is also represented in the manifest.
+    * The signature of the manifest matches the ECU key of the primary that sent it.
+    * The signature of each secondary's contribution to the manifest matches the ECU key of that secondary.
+1. The Director extracts information about currently installed images from the vehicle version manifest. Using this information, it determines which images should be installed next. The exact process by which this determination takes place is out of scope of this standard. However, it MUST take into account *dependencies* and *conflicts* between images, and SHOULD consult well-established techniques for dependency resolution.
```

Comment:
Is it perhaps worth stating here what the goal of "determining which images should be installed next" is? It isn't entirely clear that the Director should be trying to get the vehicle to a state that matches what it has been told should be on the vehicle.

Reply:
How about changing the language of the last step to this:

`The Director generates new metadata representing the desired set of images to be installed in the vehicle, based on the dependency resolution in step 4. This includes targets ({{targets_meta}}), snapshot ({{snapshot_meta}}), and timestamp ({{timestamp_meta}}) metadata. It then returns this metadata to the primary as described in {{download_meta}}.`

Reply:
Do you mean adding that as an extra step? Otherwise, aren't we taking about step 4? But I like the suggested text, other than that I'd prefer to change `returns this metadata` to `sends this metadata`.

Reply:
So, to me step 4 is about the abstract process of figuring out the manifest of software to install. Then, in the optional step 5, we encrypt any images that require it based on the map file. Finally, in step 6, based on what we've done in steps 4 and 5, we can generate and send the metadata that the primary needs.

I meant the above text to replace the text in step 6, leaving the current text of step 4 intact. Maybe I could also change the text of step 4 to read "[...] determines if the vehicle is already up-to-date, and if not, determines a set of images that should be installed. [...]" How about that?

Reply:
Ah, now I understand, that was my mistake. I like both suggestions (for steps 4 and 6). They're small, but I think they help.

Reply:
Done, pushed in a fixup commit.